src/EventStore.Core/DataStructures/UnmanagedMemoryAppendOnlyList.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace EventStore.Core.DataStructures {
+
+	public interface IAppendOnlyList<in T> {
+		void Add(T item);
+		void Clear();
+		int Count { get; }
+	}
+
+	public sealed class UnmanagedMemoryAppendOnlyList<T> : IAppendOnlyList<T>, IDisposable {
+		private readonly int _maxCapacity;
+		private readonly IntPtr _dataPtr = IntPtr.Zero;
+		private readonly long _dataBytesAllocated;
+
+		private int _count;
+		private bool _disposed;
+
+		public UnmanagedMemoryAppendOnlyList(int maxCapacity) {
+			if (maxCapacity <= 0) {
+				throw new ArgumentException("maxCapacity must be positive");
```

Comment:
Should add `nameof(maxCapacity)` to the exception's constructor

Reply:
thanks @thefringeninja! fixed in this commit: 44382425ecf32b71ba9db77eae9903197b1583ca